src/maker/config.rs

```diff
@@ -29,23 +33,251 @@ pub struct MakerConfig {
     pub min_size: u64,
 }
 
+impl Default for MakerConfig {
+    fn default() -> Self {
+        Self {
+            port: 3000,
+            heart_beat_interval_secs: 3,
+            rpc_ping_interval_secs: 60,
+            watchtower_ping_interval_secs: 300,
+            directory_servers_refresh_interval_secs: 60 * 60 * 12, //12 Hours
+            idle_connection_timeout: 300,
+            onion_addrs: "onion@example".to_string(),
+            absolute_fee_sats: 1000,
+            amount_relative_fee_ppb: 10_000_000,
+            time_relative_fee_ppb: 100_000,
+            required_confirms: 1,
+            min_contract_reaction_time: 48,
+            min_size: 10_000,
+        }
+    }
+}
+
 impl MakerConfig {
     /// Init a default configuration with given port and address
-    pub fn init(port: u16, onion_addrs: String) -> Self {
-        Self {
+    pub fn init(port: u16, onion_addrs: String, file_path: Option<&PathBuf>) -> Self {
+        let sections = if let Some(path) = file_path {
+            parse_toml(path)
+        } else {
+            parse_toml(&PathBuf::from("maker.toml"))
+        };
+
+        if sections.is_err() {
+            return Self {
+                port,
+                onion_addrs,
+                ..MakerConfig::default()
+            };
+        }
+        let section = sections.unwrap();
+
+        if let None = section.get("maker.toml") {
+            return Self {
+                port,
+                onion_addrs,
+                ..MakerConfig::default()
+            };
+        }
+
+        let maker_config_section = section.get("maker_config").unwrap();
+
+        let maker_keys = vec![
+            "absolute_fee_sats",
+            "amount_relative_fee_ppb",
+            "time_relative_fee_ppb",
+            "required_confirms",
+            "min_contract_reaction_time",
+            "min_size",
+            "idle_connection_timeout",
+            "watchtower_ping_interval_secs",
+            "heart_beat_interval_secs",
+            "heart_beat_interval_secs",
+            "rpc_ping_interval_secs",
+            "directory_servers_refresh_interval_secs",
+        ];
+
+        for x in maker_keys.iter() {
+            if maker_config_section.contains_key(*x) == false {
+                return Self {
+                    port,
+                    onion_addrs,
+                    ..MakerConfig::default()
+                };
```

Comment:
I could be reading it wrong, but it seems even if one single line is missing it will make the whole config default?

Reply:
Yes... Currently I am just taking up the default config .... if the provided config file is not proper.